include/scale/scale_decoder_stream.hpp

```diff
@@ -208,28 +220,28 @@ namespace scale {
      * @param v optional value reference
      * @return reference to stream
      */
-    template <class T>
+    template <typename T>
+      requires std::is_default_constructible_v<std::decay_t<T>>
```

Comment:
Should we mix `decay` and `remove_cvref` in `requires` and `using` (below), or choose one?

Reply:
Should not. You right